### PR TITLE
FLU: Implement backward compability mechanism

### PR DESCRIPTION
**Pull request by lukicdarkoo**
To avoid breaking worlds and PROTOs we add a mechanism to handle the backward compatibility by adding Transform nodes and rotation Transform descendants.

**Tasks**
  - [x] Proof of concept
  - Cases
    - [x] 21b PROTO inside 21b PROTO
    - [x] 21c PROTO inside 21b PROTO
    - [x] 21b PROTO inside 21c PROTO
    - [x] Primitive inside 21b PROTO
    - [x] Primitive inside 21b world
  - Cover all nodes
    - [x] Cylinder
    - [x] Capsule
    - [x] Cone
    - [x] Plane
    - [x] Elevation grid
    - [x] Camera
    - [x] Range Finder
    - [x] Lidar
    - [x] Radar
    - [x] Track
    - [x] Viewpoint
    - [x] TouchSensor
    - [x] Connector
    - [x] Emitter & Receiver
    - [x] Pen
  - [x] Fix reset 
  - [x] Handle USE/DEF
  - [x] Add tests

World example
```
#VRML_SIM R2021c utf8
WorldInfo {
}
Viewpoint {
  orientation 0.1763811647303118 -0.3150091039421665 -0.9325550649489456 5.201713829247902
  position -0.3639376985270853 -1.5105649727458803 1.0602391552040498
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 100 100
}
Robot {
  translation -2.69 0 0
  children [
    Camera {
      children [
        DEF BOX_SHAPE Shape {
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      boundingObject USE BOX_SHAPE
    }
  ]
}
TrashContainer {
}
JetBot {
}
UR3e {
  translation 0 0.33 1.54865e-06
  rotation 1 0 0 4.692820414042842e-06
}
DEF TEST Spot {
  translation 1.02 0.624 0.6
  rotation 1 0 0 1.5708
}
```

<table>
<tr><td colspan="2">Imagine a user has a custom built of the Khepera IV robot</td></tr>
<tr>
<td>

Screenshot-20210914165331-531x592
</td>
<td>

Screenshot-20210914165456-402x384
</td>
</tr>

<tr>
<td>

Screenshot-20210916145443-909x673
Before the backwards compatibility fix
</td>
<td>

Screenshot-20210916145502-986x624
After the backwards compatibility fix
</td>
</tr>

</table>

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
There are a lot of edge cases. From my point of view, it is not critical to cover all of them as far as we properly notify a user of them (and we cover like 95% of cases, otherwise it doesn't make sense).

With this PR, we can achieve "a high conversion coverage" for devices and geometries. However, we have changed the orientation of pretty much any PROTO. I am afraid a lot of users use our PROTOs (a typical example being the `Floor` PROTO), so the old worlds will be broken in any case. Maybe we can add a special tag for converted PROTOs? The tag would need to have information about a rotation we made.

**Comment by lukicdarkoo** [comment]
The tests are passing locally, the Windows error seems not to be related to the PR.

Keep in mind in mind there are probably a lot of edge cases that are not covered. However, I will believe it will significantly improve the user experience (see the examples in the original comment).

**Review by omichel — CHANGES_REQUESTED** [review]
Although, I didn't yet fully tested, it looks good to me.
There is a series of apparently unrelated files committed here however.

File: src/webots/gui/WbDragTransformEvent.cpp [hunk 1]
```diff
@@ -394,8 +394,7 @@ void WbDragRotateAroundAxisEvent::apply(const QPoint &currentMousePosition) {
   mManipulator->showRotationLine(true);
   mManipulator->updateRotationLine(mViewpoint->pick(mObjectScreenPosition.x(), mObjectScreenPosition.y(), mZEye),
                                    mViewpoint->pick(currentMousePosition.x(), currentMousePosition.y(), mZEye),
-                                   mViewpoint->orientation()->value(),
-                                   mViewpoint->viewDistanceUnscaling(detachedHandlePosition) * 9);
+                                   mViewpoint->orientation()->value(), mViewDistanceUnscaling);
```
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply src/webots/gui/WbDragTransformEvent.cpp:397]: It fixes the new object rotation, I can make a separate PR (it was just convenient to discover and implement small UX improvements).

File: src/webots/wren/WbTranslateRotateManipulator.cpp [hunk 2]
```diff
@@ -337,7 +337,7 @@ void WbTranslateRotateManipulator::updateRotationLine(const WbVector3 &begin, co
   // Scale the double arrow size
   WrViewport *viewport = wr_scene_get_viewport(wr_scene_get_instance());
   const float width = wr_viewport_get_width(viewport), height = wr_viewport_get_height(viewport),
-              maxDimension = height > width ? height : width, arrowScaleFactor = arrowScale * 2.0f / maxDimension;
+              maxDimension = height > width ? height : width, arrowScaleFactor = (arrowScale * 2.0f / maxDimension) * 10.0f;
```
[the remark this answers is not in the document]
Comment (lukicdarkoo) [reply src/webots/wren/WbTranslateRotateManipulator.cpp:340]: Also related to the object rotation

File: src/webots/gui/WbDragTransformEvent.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (omichel) [reply src/webots/gui/WbDragTransformEvent.cpp:397]: OK, no need to make a separate PR, that's fine.
